src/RadioButton.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react'
+import styled from 'styled-components'
+import PropTypes from 'prop-types'
+import { color } from 'styled-system'
+import Icon from './Icon'
+import Text from './Text'
+
+const RadioButton = ({
+  className,
+  theme,
+  onClick,
+  isSelected,
+  disabled,
+  radioButtonText
+}) => {
+  const radioIconName = isSelected ? 'radioFilled' : 'radio'
+  const radioIconColor = isSelected ? 'blue' : 'borderGray'
+
+  function handleClick() {
+    if (disabled || isSelected) return
+    onClick()
+  }
+
+  return (
+    <RadioButtonWrap
+      className={className}
+      onClick={handleClick}
+      isSelected={isSelected}
+      disabled={disabled}
+      theme={theme}
+    >
+      <RadioIconWrap>
+        <Icon name={radioIconName} size={24} color={radioIconColor} />
+      </RadioIconWrap>
+      <Text.span pl={4}>{radioButtonText}</Text.span>
+    </RadioButtonWrap>
+  )
+}
+
+const RadioButtonWrap = styled.label`
+  position: relative;
+  cursor: ${props => (props.isSelected || props.disabled ? null : 'pointer')};
+
+  font-weight: ${props =>
+    props.isSelected ? props.theme.bold : props.theme.regular};
```

Comment:
I don't know if @odmln intended the font-weight change, but since font-weight affects the width of the element, generally it's better not to change based on state